### DISCUSSION TITLE
prob01-poker question thread

### DISCUSSION THREAD

**Discussion by zmughal** [CougarCS/code-reviews#14] [Q&A]
Ask for clarifications about the problem here.

**Comment by zmughal** [comment]
@Inseighn, if you would like to ask about branching here instead of #4, it will be more visible (maybe).

**Comment by StefanoFiumara** [comment]
When checking for a straight, can an Ace be both a low card and a high card? e.g. is A-2-3-4-5 a valid straight, or can the ace only be used in a 10-J-Q-K-A sequence?

**Comment by Inseighn** [comment]
Must I create a new pull request after each commit/code change for it to be visible?

**Comment by zmughal** [comment]
@Inseighn , nope, you just keep on committing! That's the advantage of using a branch for each problem

@StefanoFiumara, according to [Wikipedia](http://en.wikipedia.org/wiki/List_of_poker_hands#Straight): "the ace may not "wrap around", or play both high and low".

**Comment by StefanoFiumara** [comment]
Right, no wrapping, but the Ace can still be the low card in a straight like A-2-3-4-5, which is a special case that I must test for..

**Comment by zmughal** [comment]
Oh, ha, I see. Didn't have my caffeine yet when I wrote that... @zeebo404, want to update the problem statement to include this?

**Comment by devzeebo** [comment]
Yes, the Ace can be low, but ONLY in straights or straight flushes (A-2-3-4-5)

**Comment by zmughal** [comment]
I updated the spec.

**Comment by StefanoFiumara** [comment]
What is the order by which suits are to be sorted? I always thought it was Hearts -> Diamonds -> Clubs -> Spades, but the problem description has them in a different order.

edit: wiki says most poker variations don't rank suits in a specific order, so this may be irrelevant.

**Comment by zmughal** [comment]
Ah, OK. :-)

I have some more data in my repo, if anyone wants to use it https://github.com/zmughal/code-reviews/blob/prob01-poker/zmughal/prob01-poker/expect.txt.

**Comment by zmughal** [comment]
Hey everyone, there's a new problem up (prob02-word-chains). You can continue to work on the poker problem of course. See it to the end!
